simple-stack-example-kotlin-masterdetail-fragments/src/main/java/com/example/ktdagger/AppModule.kt

```diff
@@ -0,0 +1,83 @@
+package com.example.ktdagger
+
+import android.content.Context
+import com.example.ktdagger.schedulers.BaseSchedulerProvider
+import com.example.ktdagger.schedulers.SchedulerProvider
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+
+import javax.inject.Singleton
+
+import dagger.Module
+import dagger.Provides
+import io.reactivex.disposables.CompositeDisposable
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+
+@Module
+class AppModule(private val context: Context) {
+
+    @Provides
+    @Singleton
+    internal fun provideApplicationContext(): Context {
```

Comment:
Please remove "internal" from here, it just pollutes the byte code but isn't the same as internal. Might as well just not have it.

Also. Recommended to remove "provide" prefix from these methods. Dagger already generates the methods that way based on the annotation.